src/App.js

```diff
@@ -1,13 +1,107 @@
-import React from 'react';
+import React from "react";
+import TodoList from "./components/TodoComponents/TodoList";
+import TodoForm from "./components/TodoComponents/TodoForm";
+
+import "./components/TodoComponents/Todo.css";
+
+const todos = [
+  {
+    task: "Organize Garage",
+    id: 1528817077286,
+    completed: false
+  },
+  {
+    task: "Bake Cookies",
+    id: 1528817084358,
+    completed: false
+  }
+];
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.state = {
+      todos: todos,
+      newTodo: "",
+      completed: false
+    };
+  }
+
+  newTodoHandler = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  addTodo = event => {
+    event.preventDefault();
+    this.setState({
+      todos: [
+        ...this.state.todos,
+        {
+          task: this.state.newTodo,
+          id: Date.now(),
+          completed: false
+        }
+      ],
+      newTodo: ""
+    });
+  };
+
+  markComplete = itemId => {
+    this.setState(prevState => {
+      return {
+        todos: prevState.todos.map(banana => {
+          if (banana.id === itemId) {
+            return {
+              task: banana.task,
+              id: banana.id,
+              completed: !banana.completed
+            };
+          } else {
+            return banana;
+          }
+        })
+      };
+    });
+  };
+
+  removeComplete = event => {
+    event.preventDefault();
+    this.setState(prevState => {
+      return {
+        todos: prevState.todos.filter(allDone => {
+          return !allDone.completed;
+        })
+      };
+    });
+  };
+
+  // formSubmitHandler = event => {
+  //   event.preventDefault();
+  //   let newTodo = {
+  //     task: this.state.task,
+  //     id: Date.now(),
+  //     completed: false
+  //   };
+  //   this.setState(prevState => {
+  //     return {
+  //       todos: [...prevState.todos, newTodo],
+  //       task: "",
+  //       id: "",
+  //       completed: ""
+  //     };
+  //   });
+  // };
+
   render() {
     return (
-      <div>
-        <h2>Welcome to your Todo App!</h2>
+      <div className="app-container">
+        <TodoList todos={this.state.todos} markComplete={this.markComplete} />
+        <TodoForm
+          addTodo={this.addTodo}
+          newTodoHandler={this.newTodoHandler}
+          newTodo={this.state.newTodo}
+          removeComplete={this.removeComplete}
+        />
```

Comment:
nice job keeping naming simple and consistent to avoid confusion. improves readability so much